snap/hooks/configure

```diff
@@ -22,11 +22,14 @@ _modify_vm_args() {
   opt=$1
   value="$2"
   replace_line="-$opt $value"
-  if $(grep -q "^[#|[:space:]]*-$opt " $VM_ARGS); then
-    sed --in-place "s/^[#|[:space:]]*-$opt .*/$replace_line/" $VM_ARGS 
+  if $(grep -q "^# -$opt" $VM_ARGS); then
+    sed --in-place "s/^# -$opt.*/$replace_line/" $VM_ARGS
+  elif $(grep -q "^-$opt " $VM_ARGS); then
+    sed --in-place "s/^-$opt .*/$replace_line/" $VM_ARGS
   else
     echo $replace_line >> $VM_ARGS
   fi
+  snapctl unset $key
```

Comment:
Looking at the `unset` docs of `snapctl`.
If we change a key, then the new value is set and the "old" $key is unset, but which state is CoucDB then? Which value us active, or is the whole configuration reloaded?

Reply:
The hook should only use the set value for the sed transformation, after which it can be discarded. Subsequent calls to `snap set` should then be faster. The ultimate state is the text in the local.ini and vm.args files at the time of the `snap restart`.